src/Ztm.Zcoin.Rpc/ZcoinRpcClient.cs

```diff
@@ -31,6 +32,10 @@ public ZcoinRpcClient(RPCClient client, ITransactionEncoder exodusEncoder)
 
             this.client = client;
             this.exodusEncoder = exodusEncoder;
+
+            this.genesisTransactions = new HashSet<uint256>(
+                this.client.Network.GetGenesis().Transactions.Select(t => t.GetHash()));
```

Comment:
This still suffer from the same problems due to `ZcoinRpcClient` will be create every time when somebody want to do RPC request.